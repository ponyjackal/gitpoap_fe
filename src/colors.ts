/* -------- Standard Colors ------------*/
export const White = '#FFFFFF';

/* ----------Legacy Colors------------- */
export const PrimaryBlue = '#307AE8';
export const PrimaryBlueHover = '#255092';
export const PrimaryBlueDark = '#174BE6';

export const Blue1 = '#6682E9';

export const DarkBlue2 = '#4F6693';

export const Slate1 = '#17457C';
export const Slate2 = '#8895AE';

export const Black1 = '#3C4B69';
export const Black2 = '#445371';

export const Gray1 = '#F8FAFF';
export const Gray2 = '#D7DDEA';
export const Gray3 = '#DCDDFA';
export const Gray4 = '#ACBED4';
export const Gray5 = '#E8EDFA';
export const Gray6 = '#E3EBF6';

/* -------- DeepWork Colors ------------*/
/* Proposition to use real color names, and use those when setting a theme */
export const DarkGray = '#393A51';
export const MidnightBlue = '#10111E';

export const BackgroundPanel = '#1E1F2E';

export const TextLight = '#E2E2EE';
export const TextAccent = PrimaryBlue;
export const TextGray = '#838495';
export const TextDarkGray = DarkGray;

export const ExtraHover = '#5596F6';
export const ExtraActive = '#1761CF';
export const ExtraPressed = '#1761CF';
