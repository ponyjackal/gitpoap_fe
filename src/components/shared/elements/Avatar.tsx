import styled from 'styled-components';
import { rem } from 'polished';

export const Avatar = styled.img`
  height: ${rem(80)};
  width: ${rem(80)};
  border-radius: 50%;
`;
