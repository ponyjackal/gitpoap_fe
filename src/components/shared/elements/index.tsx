export * from './Avatar';
export * from './Badge';
export * from './Button';
export * from './Checkbox';
export * from './ClaimCircle';
export * from './CollapsibleAddress';
export * from './CopyableText';
export * from './DateInput';
export * from './Divider';
export * from './GitPOAPBadge';
export * from './Header';
export * from './HexagonPath';
export * from './IconCount';
export * from './InfoHexBase';
export * from './Input';
export * from './InputWrapper';
export * from './Label';
export * from './Loader';
export * from './MultiSelect';
export * from './NavLink';
export * from './NumberInput';
export * from './OrgName';
export * from './Pagination';
export * from './POAPBadge';
export * from './Radio';
export * from './RelativeDate';
export * from './Select';
export * from './Share';
export * from './Skeletons';
export * from './Text';
export * from './TextArea';
export * from './Title';
export * from './TwitterShareButton';
