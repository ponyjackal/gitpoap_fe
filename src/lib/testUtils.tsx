import { MantineProvider } from '@mantine/core';
import { render, RenderOptions } from '@testing-library/react';
import { ReactElement } from 'react';
import { theme } from './theme';

const Providers = ({ children }: { children: ReactElement }) => {
  return (
    <MantineProvider theme={theme} withGlobalStyles withNormalizeCSS>
      {children}
    </MantineProvider>
  );
};

export const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: Providers, ...options });
