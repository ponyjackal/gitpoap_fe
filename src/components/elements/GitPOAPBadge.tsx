import React from 'react';
import styled from 'styled-components';
import { rem } from 'polished';
import { TextLight, MidnightBlue } from '../../colors';

type Props = {
  imgUrl: string;
};

type HexProps = {
  size: number;
  borderSize: number;
};

const Hexagon = styled.div`
  clip-path: polygon(5% 25%, 50% 0, 95% 25%, 95% 75%, 50% 100%, 5% 75%);
`;

const HexOuterBorder = styled(Hexagon)<HexProps>`
  position: relative;
  --s: ${(props) => rem(props.size + 4 * props.borderSize)};
  width: var(--s);
  height: calc(var(--s) * 1);
  background-color: ${TextLight};
`;
const HexInnerBorder = styled(Hexagon)<HexProps>`
  --s: ${(props) => rem(props.size + 2 * props.borderSize)};
  position: absolute;
  top: ${(props) => rem(props.borderSize)};
  left: ${(props) => rem(props.borderSize)};

  width: var(--s);
  height: calc(var(--s) * 1);
  background: ${MidnightBlue};
`;

const HexBadge = styled(Hexagon)<{ url: string } & HexProps>`
  --s: ${(props) => rem(props.size)};
  position: absolute;
  top: ${(props) => rem(props.borderSize)};
  left: ${(props) => rem(props.borderSize)};

  width: var(--s);
  height: calc(var(--s) * 1);
  display: inline-block;
  font-size: initial; /* we reset the font-size if we want to add some content */
  background: url('${(props) => props.url}') no-repeat center center;
  background-size: cover;
`;

export const GitPOAPBadge = ({ imgUrl }: Props) => {
  const size = 200;
  const borderSize = 4;
  return (
    <HexOuterBorder size={size} borderSize={borderSize}>
      <HexInnerBorder size={size} borderSize={borderSize}>
        <HexBadge url={imgUrl} size={size} borderSize={borderSize} />
      </HexInnerBorder>
    </HexOuterBorder>
  );
};

/* Another option is to use SVG ~ https://stackoverflow.com/questions/67458234/mask-image-with-svg-shape-and-add-a-border*/
