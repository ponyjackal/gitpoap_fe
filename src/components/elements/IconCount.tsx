import styled from 'styled-components';
import { rem } from 'polished';
import { ExtraHover, ExtraPressed, TextGray, TextLight } from '../../colors';

type Props = {
  icon: React.ReactNode;
  href?: string;
  count: number;
};

const Count = styled.span`
  font-family: PT Mono;
  font-style: normal;
  font-weight: bold;
  font-size: ${rem(12)};
  line-height: ${rem(17)};
  letter-spacing: ${rem(0.5)};
  text-transform: uppercase;
  color: ${TextLight};
  margin-left: ${rem(6)};
`;

const Container = styled.a`
  display: inline-flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;

  path {
    transition: 200ms fill ease;
    fill: ${TextGray};
  }

  &:hover {
    path {
      fill: ${ExtraHover};
    }
  }
  &:active {
    path {
      fill: ${ExtraPressed};
    }
  }
`;

export const IconCount = ({ icon, href, count }: Props) => {
  return (
    <Container href={href}>
      {icon}
      {count && <Count>{count}</Count>}
    </Container>
  );
};
