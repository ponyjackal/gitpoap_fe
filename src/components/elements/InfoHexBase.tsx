import React from 'react';
import styled from 'styled-components';
import { rem } from 'polished';
import { BackgroundPanel } from '../../colors';

type Props = {
  children: React.ReactNode;
};

const StyledInfoHex = styled.div`
  display: inline-flex;
  flex-direction: column;
  min-width: ${rem(250)};
`;

const UpperTriangle = styled.div`
  clip-path: polygon(50% 0%, 0% 100%, 100% 100%);
  width: 100%;
  height: ${rem(70)};
  background-color: ${BackgroundPanel};
`;

const LowerTriangle = styled(UpperTriangle)`
  clip-path: polygon(50% 100%, 0 0, 100% 0);
`;

const Body = styled.div`
  background-color: ${BackgroundPanel};
  min-height: ${rem(100)};
`;

export const InfoHexBase = (props: Props) => {
  return (
    <StyledInfoHex>
      <UpperTriangle />
      <Body>{props.children}</Body>
      <LowerTriangle />
    </StyledInfoHex>
  );
};
