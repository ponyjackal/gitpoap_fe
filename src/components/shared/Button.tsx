import React from 'react';
import styled from 'styled-components';
import { Button as MButton } from '@mantine/core';
import { rem } from 'polished';
import { PrimaryBlue, TextAccent } from '../../colors';

type Props = {
  children: React.ReactNode;
  className?: string;
  onClick?: React.MouseEventHandler;
  disabled?: boolean;
  leftIcon?: React.ComponentProps<typeof MButton>['leftIcon'];
};

const StyledButton = styled(MButton)`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  background-color: ${PrimaryBlue};
  font-family: 'PT Mono', monospace;
  font-weight: 700;
  color: white;
  text-transform: uppercase;
  font-size: ${rem(12)};
  line-height: ${rem(18)};
  letter-spacing: ${rem(2)};
  transition: 200ms background ease;
  border-radius: ${rem(6)};
  padding: ${rem(8)} ${rem(14)};
  height: auto;

  &:hover {
    box-shadow: none;
    background-color: ${TextAccent};
  }

  &:active {
    background-color: #20457e;
  }
`;

export const Button = ({ children, className, onClick, disabled, leftIcon }: Props) => {
  return (
    <StyledButton
      compact
      className={className}
      size="md"
      disabled={disabled}
      onClick={onClick}
      leftIcon={leftIcon}
    >
      {children}
    </StyledButton>
  );
};
