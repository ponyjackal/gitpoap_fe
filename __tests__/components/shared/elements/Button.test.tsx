import { screen } from '@testing-library/react';
import 'jest-styled-components';
import { rem } from 'polished';
import { DarkGray, PrimaryBlue, TextGray, White } from '../../../../src/colors';
import { Button } from '../../../../src/components/shared/elements';
import { customRender } from '../../../../src/lib/testUtils';

describe('Button', () => {
  it('renders a button', () => {
    customRender(<Button>Button</Button>);
    const button = screen.getByRole('button', { name: 'Button' });

    expect(button).toHaveTextContent('Button');
    expect(button).toBeInTheDocument();
    expect(button).toMatchSnapshot();
  });

  it('renders a button correctly with variant - filled', () => {
    customRender(<Button variant="filled">Button</Button>);
    const button = screen.getByRole('button', { name: 'Button' });

    expect(button).toHaveStyle(`background-color: ${PrimaryBlue}`);
    expect(button).toHaveStyle(`color: ${White}`);
    expect(button).toMatchSnapshot();
  });

  it('renders a button correctly with variant - outlined', () => {
    customRender(<Button variant="outline">Button</Button>);
    const button = screen.getByRole('button', { name: 'Button' });

    expect(button).toHaveStyle(`background-color: transparent`);
    expect(button).toHaveStyle(`border-color: ${TextGray}`);
    expect(button).toHaveStyle(`border-width: ${rem(2)}`);
    expect(button).toHaveStyle(`color: ${White}`);
    expect(button).toMatchSnapshot();
  });

  it('has the correct styles when disabled', () => {
    customRender(
      <Button disabled variant="filled">
        Button
      </Button>,
    );

    const button = screen.getByRole('button', { name: 'Button' });

    expect(button).toBeDisabled();
    expect(button).toMatchSnapshot();

    expect(button).toHaveStyle(`background-color: ${DarkGray}`);
    expect(button).toBeDisabled();
  });
});
