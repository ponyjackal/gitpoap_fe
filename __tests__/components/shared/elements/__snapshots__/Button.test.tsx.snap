// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button has the correct styles when disabled 1`] = `
.emotion-2 {
  -webkit-tap-highlight-color: transparent;
  font-family: PT Mono,monospace;
  cursor: pointer;
  border: 0;
  padding: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: 16px;
  background-color: transparent;
  text-align: left;
  color: #C1C2C5;
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  height: 36px;
  padding-left: 18px;
  padding-right: 18px;
  font-family: PT Mono,monospace;
  -webkit-tap-highlight-color: transparent;
  display: inline-block;
  width: auto;
  border-radius: 6px;
  font-weight: 600;
  position: relative;
  line-height: 1;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  border: 1px solid transparent;
  background-color: #1971c2;
  color: #fff;
  border-width: 0.125rem;
  letter-spacing: 0.125rem;
  -webkit-transition: 150ms background ease,150ms color ease,150ms border ease;
  transition: 150ms background ease,150ms color ease,150ms border ease;
  background-color: #307AE8;
}

.emotion-2:focus {
  outline-offset: 2px;
  outline: 2px solid #1c7ed6;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2:focus {
  outline-offset: 2px;
  outline: 2px solid #1c7ed6;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

@media (hover: hover) {
  .emotion-2:hover {
    background-color: #1864ab;
  }
}

@media (hover: none) {
  .emotion-2:active {
    background-color: #1864ab;
  }
}

.emotion-2:active {
  -webkit-transform: translateY(1px);
  -moz-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
}

.emotion-2[data-disabled] {
  border-color: transparent;
  background-color: #373A40;
  color: #25262b;
  cursor: not-allowed;
}

.emotion-2[data-disabled]:active {
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-2[data-loading] {
  pointer-events: none;
}

.emotion-2[data-loading]::before {
  content: "";
  position: absolute;
  top: -1px;
  left: -1px;
  right: -1px;
  bottom: -1px;
  background-color: rgba(26, 27, 30, 0.5);
  border-radius: 6px;
  cursor: not-allowed;
}

.emotion-2:disabled {
  background-color: #393A51;
}

.emotion-2:disabled .emotion-6 {
  color: #838495;
}

.emotion-2:active:not(:disabled) {
  background-color: #1761CF;
}

.emotion-2:hover:not(:disabled) {
  background-color: #5596F6;
}

.emotion-2[data-loading] {
  background-color: #307AE8;
}

.emotion-2[data-loading] .emotion-6 {
  color: #FFFFFF;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
  overflow: visible;
}

.emotion-5 {
  white-space: nowrap;
  height: 100%;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.mantine-pqd7r:disabled .emotion-6 {
  color: #838495;
}

.mantine-pqd7r[data-loading] .emotion-6 {
  color: #FFFFFF;
}

.mantine-1qagcpv:disabled .emotion-6 {
  color: #838495;
}

.mantine-1qagcpv[data-loading] .emotion-6 {
  color: #FFFFFF;
}

.mantine-1h5neuf:disabled .emotion-6 {
  color: #838495;
}

.mantine-1h5neuf[data-loading] .emotion-6 {
  color: #FFFFFF;
}

.mantine-19ayk7z:disabled .emotion-6 {
  color: #838495;
}

.mantine-1qkj6a3:disabled .emotion-6 {
  color: #838495;
}

.mantine-ysubsg:disabled .emotion-6 {
  color: #838495;
}

.mantine-hyz97v:disabled .emotion-6 {
  color: #838495;
}

<button
  class="emotion-0 emotion-1 emotion-2"
  data-button="true"
  data-disabled="true"
  disabled=""
  type="button"
>
  <div
    class="emotion-3 emotion-4"
  >
    <span
      class="emotion-5 emotion-6"
      style="text-transform: uppercase;"
    >
      Button
    </span>
  </div>
</button>
`;

exports[`Button renders a button 1`] = `
.emotion-2 {
  -webkit-tap-highlight-color: transparent;
  font-family: PT Mono,monospace;
  cursor: pointer;
  border: 0;
  padding: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: 16px;
  background-color: transparent;
  text-align: left;
  color: #C1C2C5;
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  height: 36px;
  padding-left: 18px;
  padding-right: 18px;
  font-family: PT Mono,monospace;
  -webkit-tap-highlight-color: transparent;
  display: inline-block;
  width: auto;
  border-radius: 6px;
  font-weight: 600;
  position: relative;
  line-height: 1;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  border: 1px solid transparent;
  background-color: #1971c2;
  color: #fff;
  border-width: 0.125rem;
  letter-spacing: 0.125rem;
  -webkit-transition: 150ms background ease,150ms color ease,150ms border ease;
  transition: 150ms background ease,150ms color ease,150ms border ease;
  background-color: #307AE8;
}

.emotion-2:focus {
  outline-offset: 2px;
  outline: 2px solid #1c7ed6;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2:focus {
  outline-offset: 2px;
  outline: 2px solid #1c7ed6;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

@media (hover: hover) {
  .emotion-2:hover {
    background-color: #1864ab;
  }
}

@media (hover: none) {
  .emotion-2:active {
    background-color: #1864ab;
  }
}

.emotion-2:active {
  -webkit-transform: translateY(1px);
  -moz-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
}

.emotion-2[data-disabled] {
  border-color: transparent;
  background-color: #373A40;
  color: #25262b;
  cursor: not-allowed;
}

.emotion-2[data-disabled]:active {
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-2[data-loading] {
  pointer-events: none;
}

.emotion-2[data-loading]::before {
  content: "";
  position: absolute;
  top: -1px;
  left: -1px;
  right: -1px;
  bottom: -1px;
  background-color: rgba(26, 27, 30, 0.5);
  border-radius: 6px;
  cursor: not-allowed;
}

.emotion-2:disabled {
  background-color: #393A51;
}

.emotion-2:disabled .emotion-6 {
  color: #838495;
}

.emotion-2:active:not(:disabled) {
  background-color: #1761CF;
}

.emotion-2:hover:not(:disabled) {
  background-color: #5596F6;
}

.emotion-2[data-loading] {
  background-color: #307AE8;
}

.emotion-2[data-loading] .emotion-6 {
  color: #FFFFFF;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
  overflow: visible;
}

.emotion-5 {
  white-space: nowrap;
  height: 100%;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.mantine-pqd7r:disabled .emotion-6 {
  color: #838495;
}

.mantine-pqd7r[data-loading] .emotion-6 {
  color: #FFFFFF;
}

.mantine-1qagcpv:disabled .emotion-6 {
  color: #838495;
}

.mantine-1qagcpv[data-loading] .emotion-6 {
  color: #FFFFFF;
}

.mantine-1h5neuf:disabled .emotion-6 {
  color: #838495;
}

.mantine-1h5neuf[data-loading] .emotion-6 {
  color: #FFFFFF;
}

<button
  class="emotion-0 emotion-1 emotion-2"
  data-button="true"
  type="button"
>
  <div
    class="emotion-3 emotion-4"
  >
    <span
      class="emotion-5 emotion-6"
      style="text-transform: uppercase;"
    >
      Button
    </span>
  </div>
</button>
`;

exports[`Button renders a button correctly with variant - filled 1`] = `
.emotion-2 {
  -webkit-tap-highlight-color: transparent;
  font-family: PT Mono,monospace;
  cursor: pointer;
  border: 0;
  padding: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: 16px;
  background-color: transparent;
  text-align: left;
  color: #C1C2C5;
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  height: 36px;
  padding-left: 18px;
  padding-right: 18px;
  font-family: PT Mono,monospace;
  -webkit-tap-highlight-color: transparent;
  display: inline-block;
  width: auto;
  border-radius: 6px;
  font-weight: 600;
  position: relative;
  line-height: 1;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  border: 1px solid transparent;
  background-color: #1971c2;
  color: #fff;
  border-width: 0.125rem;
  letter-spacing: 0.125rem;
  -webkit-transition: 150ms background ease,150ms color ease,150ms border ease;
  transition: 150ms background ease,150ms color ease,150ms border ease;
  background-color: #307AE8;
}

.emotion-2:focus {
  outline-offset: 2px;
  outline: 2px solid #1c7ed6;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2:focus {
  outline-offset: 2px;
  outline: 2px solid #1c7ed6;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

@media (hover: hover) {
  .emotion-2:hover {
    background-color: #1864ab;
  }
}

@media (hover: none) {
  .emotion-2:active {
    background-color: #1864ab;
  }
}

.emotion-2:active {
  -webkit-transform: translateY(1px);
  -moz-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
}

.emotion-2[data-disabled] {
  border-color: transparent;
  background-color: #373A40;
  color: #25262b;
  cursor: not-allowed;
}

.emotion-2[data-disabled]:active {
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-2[data-loading] {
  pointer-events: none;
}

.emotion-2[data-loading]::before {
  content: "";
  position: absolute;
  top: -1px;
  left: -1px;
  right: -1px;
  bottom: -1px;
  background-color: rgba(26, 27, 30, 0.5);
  border-radius: 6px;
  cursor: not-allowed;
}

.emotion-2:disabled {
  background-color: #393A51;
}

.emotion-2:disabled .emotion-6 {
  color: #838495;
}

.emotion-2:active:not(:disabled) {
  background-color: #1761CF;
}

.emotion-2:hover:not(:disabled) {
  background-color: #5596F6;
}

.emotion-2[data-loading] {
  background-color: #307AE8;
}

.emotion-2[data-loading] .emotion-6 {
  color: #FFFFFF;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
  overflow: visible;
}

.emotion-5 {
  white-space: nowrap;
  height: 100%;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.mantine-pqd7r:disabled .emotion-6 {
  color: #838495;
}

.mantine-pqd7r[data-loading] .emotion-6 {
  color: #FFFFFF;
}

.mantine-1qagcpv:disabled .emotion-6 {
  color: #838495;
}

.mantine-1qagcpv[data-loading] .emotion-6 {
  color: #FFFFFF;
}

.mantine-1h5neuf:disabled .emotion-6 {
  color: #838495;
}

.mantine-1h5neuf[data-loading] .emotion-6 {
  color: #FFFFFF;
}

<button
  class="emotion-0 emotion-1 emotion-2"
  data-button="true"
  type="button"
>
  <div
    class="emotion-3 emotion-4"
  >
    <span
      class="emotion-5 emotion-6"
      style="text-transform: uppercase;"
    >
      Button
    </span>
  </div>
</button>
`;

exports[`Button renders a button correctly with variant - outlined 1`] = `
.emotion-2 {
  -webkit-tap-highlight-color: transparent;
  font-family: PT Mono,monospace;
  cursor: pointer;
  border: 0;
  padding: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: 16px;
  background-color: transparent;
  text-align: left;
  color: #C1C2C5;
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  height: 36px;
  padding-left: 18px;
  padding-right: 18px;
  font-family: PT Mono,monospace;
  -webkit-tap-highlight-color: transparent;
  display: inline-block;
  width: auto;
  border-radius: 6px;
  font-weight: 600;
  position: relative;
  line-height: 1;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  border: 1px solid #339af0;
  background-color: transparent;
  color: #339af0;
  border-width: 0.125rem;
  letter-spacing: 0.125rem;
  -webkit-transition: 150ms background ease,150ms color ease,150ms border ease;
  transition: 150ms background ease,150ms color ease,150ms border ease;
  background-color: transparent;
  border-color: #838495;
  color: #FFFFFF;
}

.emotion-2:focus {
  outline-offset: 2px;
  outline: 2px solid #1c7ed6;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2:focus {
  outline-offset: 2px;
  outline: 2px solid #1c7ed6;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

@media (hover: hover) {
  .emotion-2:hover {
    background-color: rgba(51, 154, 240, 0.05);
  }
}

@media (hover: none) {
  .emotion-2:active {
    background-color: rgba(51, 154, 240, 0.05);
  }
}

.emotion-2:active {
  -webkit-transform: translateY(1px);
  -moz-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
}

.emotion-2[data-disabled] {
  border-color: transparent;
  background-color: #373A40;
  color: #25262b;
  cursor: not-allowed;
}

.emotion-2[data-disabled]:active {
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-2[data-loading] {
  pointer-events: none;
}

.emotion-2[data-loading]::before {
  content: "";
  position: absolute;
  top: -1px;
  left: -1px;
  right: -1px;
  bottom: -1px;
  background-color: rgba(26, 27, 30, 0.5);
  border-radius: 6px;
  cursor: not-allowed;
}

.emotion-2:disabled {
  background-color: transparent;
  border-color: #838495;
}

.emotion-2:disabled .emotion-6 {
  color: #838495;
}

.emotion-2:active:not(:disabled) {
  border-color: #1761CF;
  color: #1761CF;
}

.emotion-2:hover:not(:disabled) {
  border-color: #5596F6;
  color: #5596F6;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
  overflow: visible;
}

.emotion-5 {
  white-space: nowrap;
  height: 100%;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.mantine-pqd7r:disabled .emotion-6 {
  color: #838495;
}

.mantine-pqd7r[data-loading] .emotion-6 {
  color: #FFFFFF;
}

.mantine-1qagcpv:disabled .emotion-6 {
  color: #838495;
}

.mantine-1qagcpv[data-loading] .emotion-6 {
  color: #FFFFFF;
}

.mantine-1h5neuf:disabled .emotion-6 {
  color: #838495;
}

.mantine-1h5neuf[data-loading] .emotion-6 {
  color: #FFFFFF;
}

.mantine-799x9f:disabled .emotion-6 {
  color: #838495;
}

.mantine-799x9f[data-loading] .emotion-6 {
  color: #FFFFFF;
}

.mantine-19ayk7z:disabled .emotion-6 {
  color: #838495;
}

.mantine-1qkj6a3:disabled .emotion-6 {
  color: #838495;
}

.mantine-ysubsg:disabled .emotion-6 {
  color: #838495;
}

<button
  class="emotion-0 emotion-1 emotion-2"
  data-button="true"
  type="button"
>
  <div
    class="emotion-3 emotion-4"
  >
    <span
      class="emotion-5 emotion-6"
      style="text-transform: uppercase;"
    >
      Button
    </span>
  </div>
</button>
`;
